### ISSUE TITLE
@HostBinding not triggering Attribute Directives

### ISSUE THREAD

**Issue by emilio-martinez** [angular/angular#11332]
**I'm submitting a ...**  (check one with "x")
```
[ X ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
On "Reproduction of the problem" below, I show how I'm injecting data into a component that's resolved by ComponentFactory. Said component uses HostBinding to bind one of it's values into it's attributes, which corresponds to an attribute directive selector in a shared module. My expectation would be that once HostBinding does it's work of placing the attribute on the Host, the attribute directive logic would trigger.

The fact that the component is resolved and created dynamically and that the data is injected is just a simplified version of what I'm doing on a project, where I found this issue. Testing in other ways—dynamic or not—I've arrived at the same issue.

**Expected/desired behavior**
When an attribute is set via HostBinding, the corresponding attribute directive (if any) will be engaged and execute it's logic.

**Reproduction of the problem**
http://plnkr.co/edit/STuCXmXSZON5X9w578XK?p=preview

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.6
 
* **Browser:** all, primarily testing on Chrome 52
 
* **Language:** TypeScript X.X

**Comment by zoechi** [comment]
Adding attributes, classes, ..., or whatever a directive or component selector can match, dynamically, never causes directives and components to be added.

Dup of #6970

**Comment by emilio-martinez** [comment]
Not really. I'm adding a component dynamically, which is working just fine. Once the component is created, I'm using HostBinding to set the component's attributes, which match attribute directives, and therefore from my perspective they should be engaged.

FYI, this worked back on RC4. Didn't check if it worked on RC5.

**Comment by emilio-martinez** [comment]
To add more clarity based on the other thread you mentioned, on I was previously able to engage attribute directives in the way I described, and these were declared in the directives property of the component metadata at the time. That worked just fine. 

I hadn't ported this part of my architecture to NgModules, and I wonder if that's what causes this not to work anymore, no matter what I try—an imported module of shared directives, adding the directive to the declarations property of the NgModuel metadata.

I'm not trying to create the directive per sé. I've added it to the corresponding NgModule, and it should be available to components inside it.

**Comment by zoechi** [comment]
> I'm using HostBinding to set the component's attributes, which match attribute directives, and therefore from my perspective they should be engaged.

`HostBinding` is adding the attribute dynamically. Angular applies directives only to selectors that are added statically to the components template.   
There was no change since about 11/2015 (since I'm following Angular2 - one of the last alpha versions)

**Comment by emilio-martinez** [comment]
Thanks @zoechi. So basically attribute directives are primarily meant to be used within the template syntax, not to be resolved dynamically. Correct?

**Comment by zoechi** [comment]
@emilio-martinez exactly
